### ISSUE TITLE
[Rule request] Flag unnecessary parenthesis

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3567]
```typescript
const isParameterDecorator = (typeof parameterIndex === "number");
```

Perhaps a rule to complain about the parenthesis? To me, it looks like the writer might not have understood that they're unnecessary in those situations.

It should probably ignore expressions across lines, since some folks prefer parenthesis to make multiline expressions more clear.

**Comment by ajafff** [comment]
#3238 tried to add a configurable rule for that use case. It was rejected, because the configuration was not really understandable for someone not familiar with typescript internals.
Banning all unnecessary parens on the other hand is a bit too much.

**Comment by adidahiya** [comment]
I think this falls under the category of formatting rules I mentioned in https://github.com/palantir/tslint/issues/3592. As such, I don't think we should invest effort in it in core TSLint rules.
